Explorer/Assets/DCL/StylizedSkybox/Scripts/SkyboxTimeManager.cs

```diff
@@ -0,0 +1,181 @@
+﻿using DCL.FeatureFlags;
+using DCL.Ipfs;
+using DCL.SceneRestrictionBusController.SceneRestriction;
+using DCL.SceneRestrictionBusController.SceneRestrictionBus;
+using ECS.SceneLifeCycle;
+using JetBrains.Annotations;
+using SceneRunner.Scene;
+using UnityEngine;
+
+namespace DCL.StylizedSkybox.Scripts
+{
+    public class SkyboxTimeManager
+    {
+        private struct FeatureFlagSkyboxSettings
+        {
+            public int time;
+            public int speed;
+        }
+
+        private readonly StylizedSkyboxSettingsAsset skyboxSettings;
+        private readonly ISceneRestrictionBusController sceneRestrictionBusController;
+        private readonly FeatureFlagsCache featureFlagsCache;
+        private const float DEFAULT_SPEED = 1 * 60f; // 1 minute per second
+        [CanBeNull] private ISceneFacade scene;
+        private bool hasActiveSceneOverride = false;
+        private bool hasActiveFeatureFlagOverride = false;
+        private bool needsHierarchyUpdate = true; // Force check on the first run
+        private float speedMultiplier { get; set; } = DEFAULT_SPEED;
+
+        public float GlobalTimeOfDay { get; private set; }
+
+        public SkyboxTimeManager(StylizedSkyboxSettingsAsset skyboxSettings, IScenesCache scenesCache, ISceneRestrictionBusController sceneRestrictionBusController, FeatureFlagsCache featureFlagsCache)
+        {
+            this.skyboxSettings = skyboxSettings;
+            this.sceneRestrictionBusController = sceneRestrictionBusController;
+            this.featureFlagsCache = featureFlagsCache;
+            skyboxSettings.IsDayCycleEnabled = true;
+            skyboxSettings.TimeOfDayNormalized = StylizedSkyboxSettingsAsset.DEFAULT_TIME;
+            GlobalTimeOfDay = skyboxSettings.TimeOfDayNormalized;
+
+            scenesCache.OnCurrentSceneChanged += OnSceneChanged;
+            skyboxSettings.SkyboxTimeSourceChanged += SkyboxSettingsOnSkyboxTimeSourceChanged;
+        }
+
+        private void SkyboxSettingsOnSkyboxTimeSourceChanged(SkyboxTimeSource newSource)
+        {
+            switch (newSource)
+            {
+                case SkyboxTimeSource.GLOBAL:
+                case SkyboxTimeSource.PLAYER_FIXED:
+                case SkyboxTimeSource.FEATURE_FLAG:
+                    RestrictTimeControl(false);
+                    break;
+                case SkyboxTimeSource.SCENE_FIXED:
+                    RestrictTimeControl(true);
+                    break;
+            }
+        }
+
+        private void RestrictTimeControl(bool restrict)
+        {
+            var restrictionsAction = restrict ? SceneRestrictionsAction.APPLIED : SceneRestrictionsAction.REMOVED;
+            sceneRestrictionBusController.PushSceneRestriction(SceneRestriction.CreateSkyboxTimeLocked(restrictionsAction));
+        }
+
+        public void Update(float deltaTime)
+        {
+            UpdateGlobalTime(deltaTime);
+
+            if (needsHierarchyUpdate)
```

Comment:
Is this bool required? Did you try updating the time source hierarchy after scene has changed?